calculator.rb

```diff
@@ -0,0 +1,130 @@
+#Solution using Regular Expressions
+def number_type (num)
+  if num.scan(/\./).empty?
+    num = num.to_i
+  else
+    num = num.to_f
+  end
+  return num
+end
+
+# Solution without Regular Expressions
+# def number_type (num)
+#   if num.to_f.round - num.to_f == 0
+#     return num.to_i
+#   else
+#     return num.to_f
+#   end
+# end
+
+# def is_a_number (num)
+#   if num == num.to_i.to_s || num == num.to_f.to_s
+#     return true
+#   else
+#     return false
+#   end
+# end
+
+def add (num1, num2)
+  return "#{num1} + #{num2} = #{(num1 + num2)}"
+end
+
+def subtract (num1, num2)
+  return "#{num1} - #{num2} = #{(num1 - num2)}"
```

Comment:
This line seems to be responsible for a bug:

Please choose one operator(name or symbol): subtract
Enter the first number: 9
Enter the second number: 6
9 - 6 = 3